R/plotMoonlightMet.R

```diff
@@ -0,0 +1,135 @@
+                 values_to = "Moonlight_gene_z_score") 
+  
+  # Count total number of hypo/hyper/dual methylated CpG sites 
+  # for each oncogenic mediator
+  combined_data <- Oncogenic_mediators_methylation_summary %>% 
+    dplyr::select(-c(contains("NCG"),
+                     No_methyl_num)) %>% 
+    mutate(Total_methyl_num = rowSums(dplyr::select(.,
+                                                    contains("methyl_num")), 
+                                      na.rm = TRUE)) %>% 
+    #join with ura data
+    left_join(x = .,
+              y = ura,
+              by = c("Hugo_Symbol" = "Genes")) %>% 
+    #join with DEA data
+    left_join(x = .,
+              y = DEGs_logFC,
+              by = "Hugo_Symbol") %>% 
+    #filter data to contain only genes supplied in input
+    filter(Hugo_Symbol %in% genes)
+  
+  # Get number of Moonlight_Oncogenic_Mediator groups in subsetted data
+  driver_groups <- combined_data$Moonlight_Oncogenic_Mediator %>% 
+    unique %>% 
+    length
+  
+  # If data contains both oncogenic mediator gene types
+  if (driver_groups == 2) {
+    
+    # Create heatmap of genes grouped by oncogenic mediator types
+    met_heatmap <- combined_data %>% 
+      group_by(Moonlight_Oncogenic_Mediator) %>% 
+      heatmap(.data = .,
+              .row = Biological_process,
+              .column = Hugo_Symbol, 
+              .value = Moonlight_gene_z_score,
+              scale = "none",
+              clustering_distance_columns = "euclidean",
+              clustering_method_columns = "complete",
+              cluster_rows = FALSE) %>%
+      add_tile(Moonlight_Oncogenic_Mediator, 
+               palette = c("goldenrod2", "dodgerblue3")) %>% 
+      add_tile(logFC, 
+               palette = c("chartreuse4",  "firebrick3")) %>% 
+      add_bar(Total_methyl_num) 
+    save_pdf(met_heatmap, 
+             height = 15, 
+             width = 35, 
+             units = "cm", 
+             filename = paste(additionalFilename,
+                              "moonlight_heatmap_met.pdf", 
+                              sep = ""))
+    
+    # If data only contains one oncogenic mediator gene type
+  } else {
+    
+    # Create heatmap of genes 
+    met_heatmap <- combined_data %>% 
+      heatmap(.data = .,
+              .row = Biological_process,
+              .column = Hugo_Symbol, 
+              .value = Moonlight_gene_z_score,
+              scale = "none",
+              clustering_distance_columns = "euclidean",
+              clustering_method_columns = "complete",
+              cluster_rows = FALSE) %>% 
+      add_tile(logFC, 
+               palette = c("chartreuse4",  "firebrick3")) %>% 
+      add_bar(Total_methyl_num) 
+    save_pdf(met_heatmap, 
+             height = 15, 
+             width = 35, 
+             units = "cm", 
+             filename = paste(additionalFilename,
+                              "moonlight_heatmap_met.pdf", 
+                              sep = ""))
+    
+  }
```

Comment:
same comment as the previous functions re duplicated code - can we take the plotting out of the `if`?

Reply:
updated